src/transformState.js

```diff
@@ -5,7 +5,44 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const transformedState = state;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        addProperties(transformedState, action);
+        break;
+
+      case 'removeProperties' :
+        removeProperties(transformedState, action);
+        break;
+
+      case 'clear' :
+        clearState(transformedState);
+        break;
+
+      default :
+        return null;
+    }
+  }
+
+  return transformedState;
+}
+
+function addProperties(state, action) {
+  Object.assign(state, action.extraData);
+}
+
+function removeProperties(state, action) {
+  for (const keyToRemove in action.keysToRemove) {
+    delete state[action.keysToRemove[keyToRemove]];
+  }
```

Comment:
sorry, didn't see it last time
`keysToRemove` is an array itself, let's use another loop as well